### PR TITLE
Update horizontal momentum (explicit) tendency term

### LINKED ISSUE #1274: correct weighted interpolation for horizontal transportation in the momentum equation PR1241

### PR DESCRIPTION
**Pull request by akshaysridhar**
rhoJ Weighted interpolation in horizontal tendency 
S2.1.2 Equation (20)  
Closes #1274 
Longrun Build: https://buildkite.com/clima/climaatmos-longruns/builds/774

### CODE REVIEW

**Review by Zhengyu-Huang — COMMENTED** [review]
I confirm the code and the doc are consistent for the PR

File: src/tendencies/implicit/implicit_tendency.jl
Comment (simonbyrne) [inline src/tendencies/implicit/implicit_tendency.jl:17]:
```suggestion
function vertical_transport!(ᶜρcₜ, ᶠw, ᶜρ, ᶜρc, p, ::Val{:none})
```

**Comment by akshaysridhar** [comment]
@trontrytel @szy21 This PR introduces behaviour changes to some of the `TurbulenceConvection` cases - Could you please confirm that the [buildkite results](https://buildkite.com/clima/climaatmos-ci/builds/7097#01864bef-910c-4b84-a1d8-3676e2abbdd8) are reasonable? If so, I'll update the `ref_counter` and `mse_tables` and rebase it.

**Comment by szy21** [comment]
Is there a longrun build for this PR? Could you post the plots?

**Comment by akshaysridhar** [comment]
**Temperature**
mean_temp_sigma
**Potential Temperature**
mean_pottemp_sigma
**Zonal Vel**
mean_u_sigma

**Comment by szy21** [comment]
Thanks. I checked the moist baroclinic wave and moist held suarez in the ci and they look good to me. Anna can help check the EDMF cases. Please make sure to go over all the tests in ci to make sure the results look close enough to the current main.

**Review by szy21 — APPROVED** [review]

**Comment by bischtob** [comment]
@akshaysridharl, looks like just some MSE table updates are necessary here? Are we ready for this to get in? cc: @szy21

**Comment by akshaysridhar** [comment]
The following EDMF cases appear to have outputs that require a second review : 
(1) SCM EDMF Bomex Default
(2) SCM EDMF Bomex Constant Entrainment
(3) SCM EDMF Bomex (JFNK + IMEX)
With these checked, the regression reference counter needs to be updated, then this is ready to be merged.

**Comment by szy21** [comment]
The differences in the EDMF cases look small, but I'm not sure if I'm qualified to review EDMF changes. @trontrytel can help.

**Review by trontrytel — APPROVED** [review]
There are some differences in the time evolution of JFNK Bomex simulatios.

But this is the correct way of doing things, so LGTM!

**Pull request merged**
